File: goblinframework-api/src/main/java/org/goblinframework/api/remote/ServiceVersion.java
package org.goblinframework.api.remote;

import java.lang.annotation.*;

@Documented
@Target({ElementType.TYPE, ElementType.METHOD, ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
public @interface ServiceVersion {

  String DEFAULT_VERSION = "1.0.0";

  String version() default DEFAULT_VERSION;

  boolean enable() default true;

}

File: goblinframework-api/src/main/java/org/goblinframework/api/remote/ServiceVersion.kt
package org.goblinframework.api.remote

@MustBeDocumented
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class ServiceVersion(
    val version: String = "1.0",
    val enable: Boolean = true
)
